client/apps/edit/components/section_image_collection/index.coffee

```diff
@@ -0,0 +1,269 @@
+              className: 'bordered-input bordered-input-dark'
+              placeholder: 'Add artwork url'
+            }
+              button {
+                className: 'esic-byurl-button avant-garde-button'
+                onClick: @addArtworkFromUrl
+              }, 'Add'
+      (
+        if @state.progress
+          div { className: 'upload-progress-container' },
+            div {
+              className: 'upload-progress'
+              style: width: (@state.progress * 100) + '%'
+            }
+      )
+      (
+        if @state.images.length > 0
+          ul { className: 'esic-images-list', ref: 'images' },
+            (@state.images.map (item, i) =>
+              li { key: i },
+                if item.type is 'artwork'
+                  [
+                    div { className: 'esic-img-container' },
+                      img {
+                        src: item.image
+                        className: 'esic-artwork'
+                      }
+                    p {},
+                      strong {}, @formatArtistNames item
+                    p {},
+                      span { className: 'title' }, item.title if item.title
+                      if item.date
+                        span { className: 'date' }, ", " + item.date if item.date
+                    p {}, item.partner.name if item.partner.name
+                    button {
+                      className: 'edit-section-remove button-reset esic-img-remove'
+                      dangerouslySetInnerHTML: __html: $(icons()).filter('.remove').html()
+                      onClick: @removeItem(item)
+                    }
+                  ]
+                else
+                  [
+                    div { className: 'esic-img-container'},
+                      img {
+                        className: 'esic-image'
+                        src: if @state.progress then item.url else resize(item.url, width: 900)
+                        style: opacity: if @state.progress then @state.progress else '1'
+                      }
+                    Input {
+                      caption: item.caption
+                      images: @state.images
+                      url: item.url
+                      editing: @props.editing
+                    }
+                    button {
+                      className: 'edit-section-remove button-reset esic-img-remove'
+                      dangerouslySetInnerHTML: __html: $(icons()).filter('.remove').html()
+                      onClick: @removeItem(item)
+                      key: 2
+                    }
+                    div {
+                      dangerouslySetInnerHTML: __html: item.caption
+                      className: 'esic-caption esic-caption--display'
+                    }
+                  ]
+            )
+        else
+          ul { className: 'esic-images-list--placeholder', ref: 'images' },
+            li { className: 'esic-placeholder' }, 'Add images and artworks above'
+      )
```

Comment:
Nice! 

A random thought and definitely not for today, but I wonder how similar this file is to the ImageSet too. Similar to how we do `SectionContainer`, it might be worth a refactor sometime in the future to have an `Images` component that takes care of putting artworks & images in an array.